src/test/fuzz/coinscache_sim.cpp

```diff
@@ -148,14 +146,13 @@ class CoinsViewBottom final : public CCoinsView
 public:
     std::optional<Coin> GetCoin(const COutPoint& outpoint) const final
     {
-        // TODO GetCoin shouldn't return spent coins
-        if (auto it = m_data.find(outpoint); it != m_data.end()) return it->second;
+        if (auto it{m_data.find(outpoint)}; it != m_data.end() && !it->second.IsSpent()) return it->second;
         return std::nullopt;
     }
 
     bool HaveCoin(const COutPoint& outpoint) const final
     {
-        return m_data.count(outpoint);
+        return GetCoin(outpoint).has_value();
```

Comment:
we should be able to do this safely in a prefactor PR regardless of the other changes